back/generator/model/keywords.py

```diff
@@ -0,0 +1,179 @@
+        self.assign_to_variable = assign_to_variable
+        self.variable = variable
+        self.eol = eol
+
+    def _args_generator(self) -> None:
+        generated_list = []
+        for x in self.args:
+            generated_list.append(
+                [
+                    Token(Token.SEPARATOR, "    "),
+                    Token(Token.CONTINUATION, "..."),
+                    Token(Token.SEPARATOR, "    "),
+                    Token(Token.ARGUMENT, x),
+                    Token(Token.EOL, "\n"),
+                ]
+            )
+        return flatten(generated_list)
+
+    def _kwargs_generator(self) -> list:
+        generated_list = []
+        for x in self.kwargs.keys():
+            generated_list.append(
+                [
+                    Token(Token.SEPARATOR, "    "),
+                    Token(Token.CONTINUATION, "..."),
+                    Token(Token.SEPARATOR, "    "),
+                    Token(Token.ARGUMENT, "{}={}".format(x, self.kwargs[x])),
+                    Token(Token.EOL, self.eol),
+                ]
+            )
+        return flatten(generated_list)
+
+    def dump(self) -> KeywordCall:
+        tokens = []
+
+        if self.assign_to_variable:
+            header = [
+                Token(Token.SEPARATOR, "    "),
+                Token(Token.ASSIGN, "${{{}}}=".format(self.variable)),
+                Token(Token.SEPARATOR, "    "),
+                Token(Token.KEYWORD, self.name),
+                Token(Token.EOL, self.eol),
+            ]
+        else:
+            header = [
+                Token(Token.SEPARATOR, "    "),
+                Token(Token.KEYWORD, self.name),
+                Token(Token.EOL, self.eol),
+            ]
+        tokens = tokens + header
+        dupa = tokens + self._args_generator() + self._kwargs_generator()
+        return KeywordCall.from_tokens(tokens=dupa)
+
+
+class RWCoreImportUserVariable(KeywordCallGenerator):
+    name: str = "RW.Core.Import User Variable"
+
+    def __init__(
+        self,
+        assign_to_variable=False,
+        variable="",
+        type: str = "string",
+        pattern: str = "\w",
+        example: str = "example",
+        description: str = "description",
+        default: str = "",
+    ) -> None:
+        super().__init__(assign_to_variable, variable)
+        self.kwargs = {
+            "varname": variable,
+            "type": type,
+            "description": description,
+            "pattern": pattern,
+            "example": example,
+            "default": default,
+        }
+
+
+class RWCoreImportUserSecret(RWCoreImportUserVariable):
+    name: str = "RW.Core.Import User Secret"
+
+
+class RwCliRunCli(KeywordCallGenerator):
+    name: str = "RW.CLI.Run Cli"
+
+    def __init__(
+        self,
+        assign_to_variable=False,
+        variable="",
+        cmd: str = "string",
+        render_in_commandlist: bool = True,
+        target_service: any = None,
+        secret_file__kubeconfig: any = None,
```

Comment:
FYI - the `secret__` and `secret_file__` keyword args for the `Run CLI` keyword are dynamic kwargs. They produce a secret of the name at the end of the dunder name (kubeconfig in this case) with a secret that has the contents of the secret object passed in. Here's an example with the gcp CLI for comparison: https://github.com/runwhen-contrib/rw-cli-codecollection/blob/main/codebundles/gcloud-log-inspection/runbook.robot#L60

Not a problem while you're getting it working but keep it in mind for future iterations.

Reply:
Sure, this is definitely going to be updated the way you pointed out. I'll focus on adjusting parameters support right after delivering the remaining functionalities skeletons.